pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -146,23 +146,30 @@ static void validate(PinotQuery pinotQuery)
 
   private static void validateGroupByClause(PinotQuery pinotQuery)
       throws SqlCompilationException {
-    if (pinotQuery.getGroupByList() == null) {
-      return;
-    }
-    // Sanity check group by query: All non-aggregate expression in selection list should be also included in group
-    // by list.
-    Set<Expression> groupByExprs = new HashSet<>(pinotQuery.getGroupByList());
+    boolean hasGroupByClause = pinotQuery.getGroupByList() != null;
+    Set<Expression> groupByExprs = hasGroupByClause ? new HashSet<>(pinotQuery.getGroupByList()) : null;
```

Comment:
(minor) I feel it would be more readable if we separate the logic with and without group-by because there is no overlap between these 2 handling
```
    if (hasGroupByClause) {
      ...
    } else {
      ...
    }
```

Reply:
okay  missed this comment, let me know if we still want it .

Reply:
I prefer separating the logic for readability, but it is optional. If you prefer the current way, I can also merge as is. Just let me know

Reply:
Previously, the logic was like you are suggesting (completely inside a different function which was  called in the if block) but after some discussion with @siddharthteotia  we agreed to do it this way. Hope thats okay with you